pkg/strcmp/strcmp.go

```diff
@@ -0,0 +1,77 @@
+// Copyright 2018 The Moov Authors
+// Use of this source code is governed by an Apache License
+// license that can be found in the LICENSE file.
+
+// Package strcomp defines functions to return a percent match between two
+// strings. Many algorithms are available, but all are normalized to [0.00, 1.00]
+// for use as percentages.
+//
+// Any value outside of the normalized range represents a bug and should be fixed.
+package strcmp
+
+import (
+	"unicode/utf8"
+
+	"github.com/xrash/smetrics"
+)
+
+// JaroWinkler is a more accurate version of the Jaro algorithm. It works by boosting the
+// score of exact matches at the beginning of the strings. By doing this, Winkler says that
+// typos are less common to happen at the beginning.
+//
+// For this to happen, it introduces two more parameters: the boostThreshold and the prefixSize.
+// These are commonly set to 0.7 and 4, respectively.
+//
+// From: https://godoc.org/github.com/xrash/smetrics
+func JaroWinkler(a, b string) float64 {
+	if a == "" || b == "" {
+		return 0.00
+	}
+	return smetrics.JaroWinkler(a, b, 0.7, 4)
+}
+
+// Levenshtein is the "edit distance" between two strings. This is the count of operations
+// (insert, delete, replace) needed for two strings to be equal.
+func Levenshtein(a, b string) float64 {
```

Comment:
FYI, I changed this implementation a bit in OFAC.

https://github.com/moov-io/ofac/blob/v0.2.1/pkg/strcmp/strcmp.go#L24

Reply:
I’ll take a look at the differences.